editor/plugins/visual_shader_editor_plugin.cpp

```diff
@@ -817,40 +1104,307 @@ VisualShaderEditor::VisualShaderEditor() {
+	add_options.push_back(AddOption("Specular", "Input", "Light", "VisualShaderNodeInput", "'specular' input parameter for light shader mode.", "specular", VisualShader::TYPE_LIGHT));
+	add_options.push_back(AddOption("Transmission", "Input", "Light", "VisualShaderNodeInput", "'transmission' input parameter for light shader mode.", "transmission", VisualShader::TYPE_LIGHT));
+	add_options.push_back(AddOption("View", "Input", "Light", "VisualShaderNodeInput", "'view' input parameter for light shader mode.", "view", VisualShader::TYPE_LIGHT));
+
+	add_options.push_back(AddOption("Alpha", "Input", "Vertex", "VisualShaderNodeInput", "'alpha' input parameter for vertex shader mode.", "alpha", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("Binormal", "Input", "Vertex", "VisualShaderNodeInput", "'binormal' input parameter for vertex shader mode.", "binormal", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("Color", "Input", "Vertex", "VisualShaderNodeInput", "'color' input parameter for vertex shader mode.", "color", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("ModelView", "Input", "Vertex", "VisualShaderNodeInput", "'modelview' input parameter for vertex shader mode.", "modelview", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("PointSize", "Input", "Vertex", "VisualShaderNodeInput", "'point_size' input parameter for vertex shader mode.", "point_size", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("Tangent", "Input", "Vertex", "VisualShaderNodeInput", "'tangent' input parameter for vertex shader mode.", "tangent", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("UV", "Input", "Vertex", "VisualShaderNodeInput", "'uv' input parameter for vertex shader mode.", "uv", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("UV2", "Input", "Vertex", "VisualShaderNodeInput", "'uv2' input parameter for vertex shader mode.", "uv2", VisualShader::TYPE_VERTEX));
+	add_options.push_back(AddOption("Vertex", "Input", "Vertex", "VisualShaderNodeInput", "'vertex' input parameter for vertex shader mode.", "vertex", VisualShader::TYPE_VERTEX));
+
+	// SCALAR
+
+	add_options.push_back(AddOption("ScalarFunc", "Scalar", "Common", "VisualShaderNodeScalarFunc", "Scalar function."));
+	add_options.push_back(AddOption("ScalarOp", "Scalar", "Common", "VisualShaderNodeScalarOp", "Scalar operator."));
+
+	add_options.push_back(AddOption("Abs", "Scalar", "Functions", "VisualShaderNodeScalarFunc", "Returns the absolute value of the parameter.", VisualShaderNodeScalarFunc::FUNC_ABS));
+	add_options.push_back(AddOption("ACos", "Scalar", "Functions", "VisualShaderNodeScalarFunc", "Returns the arc-cosine of the parameter.", VisualShaderNodeScalarFunc::FUNC_ACOS));
+	add_options.push_back(AddOption("ACosH", "Scalar", "Functions", "VisualShaderNodeScalarFunc", "(GLES3+)Returns the inverse hyperbolic cosine of the parameter.", VisualShaderNodeScalarFunc::FUNC_ACOSH));
```

Comment:
Maybe write `(GLES3 only)` instead of `(GLES3+)`, as we currently don't have any higher-end renderers. Make sure to add a space after the closing parenthesis as well.

Reply:
Done

Reply:
Added space too, sorry my mind is a bit reactive today )